x-pack/plugins/endpoint/server/handlers/response_to_endpoint_mapper.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { EndpointData } from '../types';
+
+interface HitSource {
+  _source: EndpointData;
+}
+
+export class ResponseToEndpointMapper {
+  async mapHits(searchResponse: SearchResponse<EndpointData>): EndpointData[] {
+    return searchResponse.hits.hits.map(response => response._source);
+  }
+
+  async mapInnerHits(searchResponse: SearchResponse<EndpointData>): EndpointData[] {
```

Comment:
same, `SearchResponse` is already the json payload, so not working with promises at this point